fix - increment number of tests run and failed

jacksonpradolima opened the pull request:
Hello,
I downloaded the **DSpace** (a Java project) *build log* file from here: https://travistorrent.testroots.org/buildlogs/20-12-2016/
Then, I run the **buildlog_analysis.rb**.
I observed that the number of tests run and failed were wrong. For instance, the file *1006_32539584_a1dffc6a2e970c45e7df09c1f4c898c26a95026d_32539585.log*, in **buildlog-data-travis.json**:
- tr_build_id: "32539584",
- tr_build_number: "1006",
- tr_original_commit: "a1dffc6a2e970c45e7df09c1f4c898c26a95026d"

The number of *ok* and *run* tests were **19**, however the total of tests found in the file are **1353**. Futhermore, I observed that the file had tests failed, and in the log the number was 0 (zero).
I changed the code and the number now is correct: 1338 (ok), 15 (failed), 1353 (run).

Please, check if my fix code is ok because I don't have knowlodge about Ruby.

In addition, I observed that the failed test names are strange. In the case of the log mentioned:
"tr_log_tests_failed": "DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#DOIIdentifierProviderTest#",

The correct would be:
- DOIIdentifierProviderTest.testStore_DOI_as_item_metadata
- DOIIdentifierProviderTest.testGet_DOI_out_of_item_metadata
- DOIIdentifierProviderTest.testRemove_DOI_from_item_metadata
- DOIIdentifierProviderTest.testGet_DOI_by_DSpaceObject
- DOIIdentifierProviderTest.testGet_DOI_lookup
- DOIIdentifierProviderTest.testGet_DSpaceObject_by_DOI
- DOIIdentifierProviderTest.testResolve_DOI
- DOIIdentifierProviderTest.testRemove_two_DOIs_from_item_metadata
- DOIIdentifierProviderTest.testMintDOI
- DOIIdentifierProviderTest.testMint_returns_existing_DOI
- DOIIdentifierProviderTest.testReserve_DOI
- DOIIdentifierProviderTest.testRegister_unreserved_DOI
- DOIIdentifierProviderTest.testRegister_reserved_DOI
- DOIIdentifierProviderTest.testCreate_and_Register_DOI
- DOIIdentifierProviderTest.testDelete_specified_DOI

In this case, the method name are not been extracted. The correct is only the test name without the method name? If yes, I guess we can leave to the developer remove or not the name.

jacksonpradolima committed:
fix - increment number of tests run and failed

Inventitech commented:
Hi, @jacksonpradolima Thanks for this PR! Unfortunately, it breaks some existing tests, as you can see in the [Travis output.](https://travis-ci.org/TestRoots/travistorrent-tools/builds/442175920?utm_source=github_status&utm_medium=notification)

The best way forward would be to add the buildlogs you are referring as example tests, just like the existing tests. Then develop a fix from there while not breaking tests.

Would you be interested in doing that?

jacksonpradolima committed:
undo java-ant increment number of tests run and failed

jacksonpradolima committed:
add DSpace logs

jacksonpradolima committed:
fix - show test method name

jacksonpradolima committed:
add test DSpace

jacksonpradolima committed:
fix test - round log test duration

jacksonpradolima commented:
Hello @Inventitech ,
I changed the code to show the method name, and I updated the tests. However when I run the command: bundle exec rake, I have the result
 337.38999999999993 in tr_log_testduration and in the Travis CI the result is 337.3899999999999.
As I said, I don't have Ruby skills, and so, I don't no why this difference.

Inventitech commented:
Problem was a version incompatibility which I could replicate locally. Merged your changes, thanks again.
